CRM/Core/BAO/CustomValueTable.php

```diff
@@ -131,13 +131,13 @@ public static function create($customParams, $parentOperation = NULL) {
                   $value = NULL;
                 }
               }
-              elseif ($value == NULL || $value === '') {
-                $type = 'Timestamp';
-                $value = NULL;
-              }
               else {
                 $type = 'Integer';
               }
+              if ($value == NULL || $value === '') {
```

Comment:
The condition could be reduced to `$value == NULL` since `'' == NULL` is true. Maybe it should be `$value === NULL`?

Reply:
Good catch! (I didn't add that line in this PR, just moved it, but yes you're right).

Reply:
@dontub https://github.com/civicrm/civicrm-core/pull/31448